File: softWrench.sW4/Data/Search/QuickSearch/QuickSearchDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using JetBrains.Annotations;

namespace softWrench.sW4.Data.Search.QuickSearch {

    public class QuickSearchDTO {

        /// <summary>
        /// List of composition (by their relationship), that should be included on the quicksearch. 
        /// The search would be performed expecting the quickSearchData to be found within any of the list fields of the composition
        /// 
        /// List instead of set, cause .net doesnt translate sets automatically 
        /// </summary>
        [CanBeNull]
        public List<string> CompositionsToInclude {
            get; set;
        }

        /// <summary>
        /// 
        /// </summary>
        [CanBeNull]
        public List<string> HiddenFieldsToInclude {
            get; set;
        }

        [NotNull]
        public string QuickSearchData {
            get; set;
        }

        public static QuickSearchDTO Basic(string toSearch) {
            return new QuickSearchDTO {
                QuickSearchData = toSearch
            };
        }

    }
}

File: softwrench.sw4.webcommons/web_content/webcommons/scripts/softwrench/dto/quickSearchDTO.js
/**
 * Clone of server side class
 * 
*/
class QuickSearchDTO {
    /**
     * 
     * @param {String} quickSearchData free text string to search for
     * @param {Array} compositionsToInclude 
     * @param {Array} compositionsToInclude 
     * 
     * @returns {} 
     */
    constructor(quickSearchData, compositionsToInclude, hiddenFieldsToInclude) {
        this.quickSearchData = quickSearchData;
        this.compositionsToInclude = compositionsToInclude;
        this.hiddenFieldsToInclude = hiddenFieldsToInclude;
    }


}


